src/Components/GEOSldas_GridComp/GEOSlandassim_GridComp/GEOS_LandAssimGridComp.F90

```diff
@@ -1146,10 +1146,51 @@ subroutine Initialize(gc, import, export, clock, rc)
          rc=status                                                              &
          )
     _VERIFY(status)
-    
+
     call ESMF_TimeIntervalSet(LandAssim_DT, s=LandAssimDtStep, rc=status)
     _VERIFY(status)
+    
+    _ASSERT(mod(LandAssimDtStep,model_dtstep)==0, "inconsistent inputs for HEARTBEAT_DT and LANDASSIM_DTSTEP")
+
+    call ESMF_TimeIntervalGet(LandAssim_DT, H = h, m = m, s = s, rc=status)
+    _VERIFY(status)
+
+    ! read reference time for land analysis from resource file (default: LandAssim_DT=3z)
+    ! ref_time is hhmmss format
+    hhmmss = h*10000 + m*100 + s
```

Comment:
The default for ref_time should simply be 000000 (ie, 0)

Reply:
Again, my definition of the ref_time is a reference time of the program starting time. It should be in hhmmss format but the total seconds is less than LandassimDTstep. I am open to different definitions

Reply:
I think I see what you're trying to do.  That is, limit the allowable rc input options for ref_time input to fall within the interval [0, LandAssimDtStep].  Put differently, users need to pick a ref_time that assumes a 0z restart and then specify the first AssimTime for the 0z restart case. 
I think this is too complicated to explain in the rc file.  If a user restarts at 12z and wants a land analysis at 13:30z, they're going to have a hard time figuring out that they need to specify 01:30z for the ref_time.   I still think we need to do the modulus calc within the src code to allow the user so specify any ref_time that's spaced LandAssimDtStep apart from the desired smallest AssimTime_of_day.   I think it's not prohibitively difficult to implement.  The pseudocode in my earlier comment should work.

Reply:
The Offset time is 5400 s and doesn't need to be in the format of hhmmss